lib/tasks/rspec.rake

```diff
@@ -7,12 +7,33 @@ namespace :knapsack_pro do
     KnapsackPro::Runners::RSpecRunner.run(args[:rspec_args])
   end
 
-  desc "Generate JSON report for test suite based on default test pattern or based on defined pattern with ENV vars"
+  # private
   task :rspec_test_example_detector do
-    # ignore the `SPEC_OPTS` options to not affect RSpec execution within this rake task
+    key = 'KNAPSACK_PRO_RSPEC_OPTIONS'
+    raise "The internal #{key} environment variable is unset. Ensure it is not overridden accidentally. Otherwise, please report this as a bug: #{KnapsackPro::Urls::SUPPORT}" if ENV[key].nil?
+
+    # Ignore `SPEC_OPTS` to not affect the RSpec execution within this rake task
     ENV.delete('SPEC_OPTS')
 
-    detector = KnapsackPro::TestCaseDetectors::RSpecTestExampleDetector.new
-    detector.generate_json_report(ENV['KNAPSACK_PRO_RSPEC_OPTIONS'])
+    KnapsackPro::TestCaseDetectors::RSpecTestExampleDetector
+      .new
+      .calculate(ENV[key])
+  end
+
+  namespace :rspec do
+    desc 'Precalculate split by test examples into KNAPSACK_PRO_RSPEC_SPLIT_BY_TEST_EXAMPLES_FILE (to avoid doing it later in each node invoking Knapsack Pro)'
+    task :precalculate_split_by_test_examples, [:rspec_args] do |_, args|
+      key = 'KNAPSACK_PRO_RSPEC_SPLIT_BY_TEST_EXAMPLES_FILE'
+      raise "Missing #{key}" if ENV[key].nil?
```

Comment:
suggestion

We could link to docs here once they include `KNAPSACK_PRO_RSPEC_SPLIT_BY_TEST_EXAMPLES_FILE`.